lib/lighthouse/benefits_claims/configuration.rb

```diff
@@ -83,19 +95,27 @@ def get_access_token?
       !use_mocks? || Settings.betamocks.recording
     end
 
-    def access_token
-      token_service.get_token if get_access_token?
+    def access_token(lighthouse_client_id, lighthouse_rsa_key_path, options = {})
+      if get_access_token?
+        token_service(
+          lighthouse_client_id,
+          lighthouse_rsa_key_path,
+          options[:aud_claim_url],
+          options[:host]
+        ).get_token(options[:auth_params])
+      end
     end
 
     ##
     # @return [BenefitsClaims::AccessToken::Service] Service used to generate access tokens.
     #
-    def token_service
-      url = "#{settings.host}/#{TOKEN_PATH}"
-      token = settings.access_token
+    def token_service(lighthouse_client_id, lighthouse_rsa_key_path, aud_claim_url = nil, host = nil)
+      host ||= base_path(host)
+      url = "#{host}/#{TOKEN_PATH}"
+      aud_claim_url ||= settings.access_token.aud_claim_url
 
       @token_service ||= Auth::ClientCredentials::Service.new(
-        url, API_SCOPES, token.client_id, token.aud_claim_url, token.rsa_key
+        url, API_SCOPES, lighthouse_client_id, aud_claim_url, lighthouse_rsa_key_path
```

Comment:
Have you tested this to make sure that this doesn't cache the first instance of `Auth::ClientCredentials::Service` that is created? Looking at this it looks like it will only ever use the first set of credentials presented to it

Reply:
@jerekshoe we have since decided to realign this code with Lighthouse's policy of only using one ClientId and RSA Key for all VA.gov; so the complexity you flagged here will be removed. So I don't think we need to investigate or solve cache busting here now.

I will skip this this question in your PR as it will be resolved by a followup ticket next Sprint or so. Let me know if you have any other concerns there.